### ISSUE TITLE
mod_admin_extra should not work with Mnesia tables directly

### ISSUE THREAD

**Issue by zinid** [processone/ejabberd#2448]
The module `mod_admin_extra` makes direct Mnesia calls to the following tables:
- `last_activity`
- `roster`
- `session`

This should be replaced with the corresponding functions from `mod_last`, `mod_roster` and `ejabberd_sm` correspondingly. If such API functions don't exist they should be written. No freaking hacks please.

**Comment by badlop** [comment]
Ok, I've improved code of connected_users_info and user_sessions_info. Once I have a working SQL testig database again, I'll update num_active_users, that requires new API. On the other hand, I think process_rosteritems is really useful only for admins with Mnesia, as SQL admins already have well-known SQL tools.

**Comment by zinid** [comment]
> On the other hand, I think process_rosteritems is really useful only for admins with Mnesia, as SQL admins already have well-known SQL tools.

I'm not sure, because hacking SQL tables requires a knowledge of the content of those SQL tables. Furthermore, the content is a subject to change quite frequently, thus admins cannot reuse their scripts safely.
Anyway, even if this is the case, it's still **wrong** to have mnesia calls to the tables outside of the "table owner" (mod_roster_mnesia in our case).
I also grepped a bit:
```
$ grep -l 'mnesia:' src/*.erl
...
src/ejabberd_web_admin.erl
src/mod_admin_extra.erl
src/mod_muc_admin.erl
...
```
These modules don't "own" any mnesia table, but make direct calls to them. This is plain wrong and significantly complicates code refactoring.

**Comment by zinid** [comment]
> On the other hand, I think process_rosteritems is really useful only for admins with Mnesia, as SQL admins already have well-known SQL tools.

By the way, there is also Riak backend and we might have more backends in the future releases, and those backends are not easy to hack by hands.

**Comment by badlop** [comment]
mod_muc_admin called two mnesia tables just to offer a pair of statistics in the web. I see no simple way to accomplish that in DB-agnostic, so I've removed the code.

mod_admin_extra had calls to the last_activity mnesia table for a statistical command called num_active_users. There is no simple way to implement that in DB-agnostic, it has the problems mentioned in previous comments, I wrote that code more than five years ago just for fun, and in fact I doubt the command is really useful, so I've removed it.

ejabberd_web_admin has similar calls to the last_activity mnesia table, used for the web page "Last activity". It was written by Aleksey, probably in the first year of ejabberd. I wonder if it's really useful, considering it works only with mnesia's mod_offline, but I won't remove it.

The only remaining mnesia calls are the ones used for the process_rosteritems in mod_admin_extra, which I'm still deciding if it's worth improving or better delete.
